go/vt/vtgate/planner.go

```diff
@@ -102,6 +102,7 @@ func (plr *Planner) WatchVSchema(ctx context.Context) {
 		plr.mu.Lock()
 		plr.vschema = vschema
 		plr.mu.Unlock()
+		plr.plans.Clear()
```

Comment:
It's still possible that a builder starts using the old vschema, and we update and clear the plan cache, and then the built plan (based on the older vschema) gets added to the cache. This is because we don't hold the lock on vschema while building the plan. Some brainstorming will be needed.